core/products/perpetual.go

```diff
@@ -164,18 +164,15 @@ type cachedTWAP struct {
 	end         int64        // time of the last calculated sub-product that was >= the last added data-point
 	sumProduct  *num.Uint    // the sum-product of all the intervals between the data-points from `start` -> `end`
 	points      []*dataPoint // the data-points used to calculate the twap
-
-	auctions *auctionIntervals
```

Comment:
Before you get too far I'd like to understand why this needs to be removed? As far as I understand we only need a few lines in `func (p *Perpetual) calculateFundingPayment(t int64)` to do:
```
factor := (t - p.startedAt) /  p.internalTWAP.auctions.timeSpent(p.startedAt, t)
```
and thats basically it?

Reply:
The formula is slightly different, but yes. 

I couldn't understand why there's a need to duplicate auction intervals withing `cachedTWAP`, thought it's just a leftover and since no tests failed after that initial refactoring I assumed I was correct. Is that not the case?

Reply:
They aren't copies its just a point, and its shared expressly to avoid having to pass it around everywhere and muddy up function signatures.

Reply:
Fair enough, I'll revert it then.